packages/main-api/src/controllers/v2/conversations/messages/poll_messages.rs

```diff
@@ -0,0 +1,111 @@
+use bdk::prelude::*;
+use dto::{
+    ConversationParticipant, Error, Message, Result,
+    by_axum::{
+        auth::Authorization,
+        axum::{
+            Extension, Json,
+            extract::{Query, State},
+        },
+    },
+    sqlx::PgPool,
+};
+use serde::{Deserialize, Serialize};
+use tokio::time::{Duration, timeout};
+
+use crate::utils::users::extract_user_id;
+
+#[derive(Debug, Clone, Serialize, Deserialize, aide::OperationIo, JsonSchema)]
+pub struct PollMessagesQuery {
+    #[schemars(description = "Conversation ID to poll messages for")]
+    pub conversation_id: i64,
+
+    #[schemars(description = "Timestamp to get messages since (in milliseconds)")]
+    pub since: Option<i64>,
+
+    #[schemars(description = "Maximum time to wait for new messages (seconds, default: 30)")]
+    pub timeout_seconds: Option<u64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, aide::OperationIo, JsonSchema)]
+pub struct PollMessagesResponse {
+    pub messages: Vec<Message>,
+    pub has_new_messages: bool,
+}
+
+pub async fn poll_messages_handler(
```

Comment:
What is the maximum timeout option for poll logic? 
I don't think it can be set indefinitely, so a max limit seems necessary.